### PR TITLE
#6 Containerization of the Database

### PR DESCRIPTION
**Pull request by Jeyashree46**
[Story link](https://tree.taiga.io/project/zubae-payment-gateway/epic/2)
 
## Description
- Pulled MongoDB image from the docker hub
- Added docker-compose file

## Checklist before requesting a review
- [x] Added docker-compose file

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+COPY . /app 
+
+WORKDIR /app 
+
+RUN go build -o main .
+
+EXPOSE 8080
+
+CMD ["/app/main"]
```
Comment (afrze) [inline Dockerfile:13]: Add an empty line to avoid EOF errors

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+services:
+  identity_service:
+    build: .
+    ports:
+      - "8080:8080"
+  mongodb:
+    image: mongo
+    ports:
+      - '2717:27017'
```
Comment (afrze) [inline docker-compose.yml:11]: Add an empty line to avoid EOF errors

**Review by afrze — CHANGES_REQUESTED** [review]
There are more tasks to be done
-  Added config which connects to the db
-  Handle errors if db connection is not going according to plan.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,13 @@
+FROM golang:latest 
+
+WORKDIR /app 
+
+COPY . /app 
+
+WORKDIR /app
```
Comment (raam-techie) [inline Dockerfile:7]: Remove line 7 WORKDIR /app [can perform all the necessary operations within a single working directory]

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,13 @@
+FROM golang:latest
```
Comment (raam-techie) [inline Dockerfile:1]: Please specify the Go version used in our application

File: docker-compose.yml (hunk 2 above)
Comment (raam-techie) [inline docker-compose.yml:11]: To keep 27017 instead of 2717

**Review by raam-techie — APPROVED** [review]

**Pull request closed without merging**